miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/js/MiniAppCustomPermissionCache.kt

```diff
@@ -0,0 +1,45 @@
+package com.rakuten.tech.mobile.miniapp.js
+
+import android.content.Context
+import android.content.SharedPreferences
+import androidx.annotation.VisibleForTesting
+import com.rakuten.tech.mobile.miniapp.js.MiniAppCustomPermissionResult.CUSTOM_PERMISSION_DENIED
+
+/**
+ * A class to check if a requested permission has been granted or rejected with storing
+ * the values of the grant result per permission.
+ */
+internal class MiniAppCustomPermissionCache(context: Context) {
+    @VisibleForTesting
+    val prefs: SharedPreferences = context.getSharedPreferences(
+        "com.rakuten.tech.mobile.miniapp.permissions", Context.MODE_PRIVATE
+    )
+
+    /**
+     * Reads grant result from SharedPreferences.
+     * @param [permissions] the key-set provided to find the stored corresponding grant results.
+     * @return [List<String>] the grant results of permissions requested.
+     */
+    fun readPermissions(permissions: List<String>): List<String> {
```

Comment:
Can this use the `MiniAppCustomPermissionType` enum instead? i.e. `permissions: List<MiniAppCustomPermissionType>`?

This way we can only allow the Host App to set valid permissions.